t/unit/tasks/test_canvas.py

```diff
@@ -1050,6 +1083,117 @@ def chord_add():
             _state.task_join_will_block = fixture_task_join_will_block
             result.task_join_will_block = fixture_task_join_will_block
 
+    def test_from_dict(self):
+        header = self.add.s(1, 2)
+        original_chord = chord(header=header)
+        rebuilt_chord = chord.from_dict(dict(original_chord))
+        assert isinstance(rebuilt_chord, chord)
+
+    def test_from_dict_with_body(self):
+        header = body = self.add.s(1, 2)
+        original_chord = chord(header=header, body=body)
+        rebuilt_chord = chord.from_dict(dict(original_chord))
+        assert isinstance(rebuilt_chord, chord)
+
+    def test_from_dict_deep_deserialize(self, subtests):
+        header = body = self.add.s(1, 2)
+        original_chord = chord(header=header, body=body)
+        serialized_chord = json.loads(json.dumps(original_chord))
+        deserialized_chord = chord.from_dict(serialized_chord)
+        with subtests.test(msg="Verify chord is deserialized"):
+            assert isinstance(deserialized_chord, chord)
```

Comment:
I was unsure if this line should be part of the main test or a subtest. It's a prerequisite for the other subtests, so I feel like it might be better to take this out of a subtest block (and similarly in the tests below).

Reply:
Generally speaking if we follow a variation of the `AAA` pattern, every assert should be in a different subtest.

I'm not going to enforce that for the time being.

Reply:
All good, sounds like I've done the right thing then for these tests with multiple assertions. `test_freeze_tasks_body_is_group()` has a single sanity assertion in the main body of the test but I'm okay with that structure since the rest of the test is nonsense if that is untrue.